tests/test_opds_import.py

```diff
@@ -748,6 +757,51 @@ def test_extract_messages(self):
         eq_(404, message.status_code)
         eq_("I've never heard of this work.", message.message)
 
+    def test_extract_medium(self):
+        m = OPDSImporter.extract_medium
+
+        # No tag -- the default is used.
+        eq_("Default", m(None, "Default"))
+
+        def medium(additional_type, format, default="Default"):
+            # Make an <atom:entry> tag with the given tags.
+            # Parse it and call extract_medium on it.
+            entry= '<entry xmlns:schema="http://schema.org/" xmlns:dcterms="http://purl.org/dc/terms/"'
+            if additional_type:
+                entry += ' schema:additionalType="%s"' % additional_type
+            entry += '>'
+            if format:
+                entry += '<dcterms:format>%s</dcterms:format>' % format
+            entry += '</entry>'
+            tag = etree.parse(StringIO(entry))
+            return m(tag.getroot(), default=default)
+
+        audio_type = random.choice(MediaTypes.AUDIOBOOK_MEDIA_TYPES) + ";param=value"
+        ebook_type = random.choice(MediaTypes.BOOK_MEDIA_TYPES) + ";param=value"
+
+        # schema:additionalType is checked first. If present, any
+        # potentially contradictory information in dcterms:format is
+        # ignored.
+        eq_(
+            Edition.AUDIO_MEDIUM,
+            medium("http://bib.schema.org/Audiobook", ebook_type)
+        )
+        eq_(
+            Edition.BOOK_MEDIUM,
+            medium("http://schema.org/EBook", audio_type)
+        )
+
+        # When schema:additionalType is missing or not useful, the
```

Comment:
In what case would it not be useful?

Reply:
It might be some random URI that we can't map to a medium.